components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/XMLSchemaValidator.java

```diff
@@ -0,0 +1,277 @@
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.EXTERNAL_ENTITIES_ENABLED);
+        if (messageProperty != null) {
+            externalEntitiesEnabled = Boolean.valueOf(messageProperty.toString());
+        } else {
+            String message = "XML schema externalEntitiesEnabled property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.MAX_ELEMENT_COUNT);
+        if (messageProperty != null) {
+            elementCount = Integer.parseInt(messageProperty.toString());
+        } else {
+            String message = "XML schema elementCount property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.MAX_ATTRIBUTE_LENGTH);
+        if (messageProperty != null) {
+            attributeLength = Integer.parseInt(messageProperty.toString());
+        } else {
+            String message = "XML schema maxAttributeLength property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.MAX_XML_DEPTH);
+        if (messageProperty != null) {
+            maxXMLDepth = Integer.parseInt(messageProperty.toString());
+        } else {
+            String message = "XML schema xmlDepth property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.MAX_ATTRIBUTE_COUNT);
+        if (messageProperty != null) {
+            attributeCount = Integer.parseInt(messageProperty.toString());
+        } else {
+            String message = "XML schema attributeCount property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.ENTITY_EXPANSION_LIMIT);
+        if (messageProperty != null) {
+            entityExpansionLimit = Integer.parseInt(messageProperty.toString());
+
+        } else {
+            String message = "XML schema entityExpansionLimit property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        }
+
+        messageProperty = messageContext.getProperty(ThreatProtectorConstants.CHILDREN_PER_ELEMENT);
+        if (messageProperty == null) {
+            String message = "XML schema childrenElement property value is missing.";
+            ThreatExceptionHandler.handleException(messageContext, message);
+        } else {
+            childrenPerElement = Integer.parseInt(messageProperty.toString());
+        }
+
```

Comment:
Should we need to  throw exception, when the relevant propertu is missing. Can't we use a default value if it is missing, without throwing an exception? May be we can include a debug log saying, "this property is missing, hence taking default value"

Reply:
yes in here we need to throw and exception.